pkg/gardenlet/operation/botanist/advertisedaddresses_test.go

```diff
@@ -195,10 +214,126 @@ var _ = Describe("AdvertisedAddresses", func() {
 			}
 			botanist.APIServerAddress = "bar.foo"
 
-			addresses, err := botanist.ToAdvertisedAddresses()
+			addresses, err := botanist.ToAdvertisedAddresses(ctx)
 			Expect(err).To(HaveOccurred())
 			Expect(err.Error()).To(Equal("shoot requires managed issuer, but gardener does not have shoot service account hostname configured"))
 			Expect(addresses).To(BeNil())
 		})
 	})
+
+	Describe("#GetIngressAdvertisedEndpoints", func() {
+		It("returns none with no ingress resources", func() {
+			items, err := botanist.GetIngressAdvertisedEndpoints(ctx)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(items).To(HaveLen(0))
+		})
+
+		It("returns none when no ingress is labeled", func() {
+			// Resource does not have the expected labels
+			ingress := &networkingv1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "ingress-1",
+					Namespace: shootNamespace.Name,
+				},
+				Spec: networkingv1.IngressSpec{
+					TLS: []networkingv1.IngressTLS{
+						{
+							Hosts: []string{"foo.example.org"},
+						},
+					},
+				},
+			}
+
+			Expect(botanist.SeedClientSet.Client().Create(ctx, ingress)).To(Succeed())
+			items, err := botanist.GetIngressAdvertisedEndpoints(ctx)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(items).To(HaveLen(0))
+			Expect(botanist.SeedClientSet.Client().Delete(ctx, ingress)).To(Succeed())
```

Comment:
See above: You don't need to do this with the fake client. It is reset for each test anyways. Otherwise, `DeferCleanup` should be used.

Reply:
Fixed.